service/src/main/java/uk/gov/hmcts/reform/fpl/model/CaseData.java

```diff
@@ -335,6 +335,14 @@ public DynamicList buildRespondentDynamicList() {
     private final DynamicList applicantsList;
     private final String otherApplicant;
 
+    // addCaseFlag event fields
+    private final DocumentReference redDotAssessmentForm;
+    private final String caseFlagNotes;
+    private final String caseFlagAdded;
+    private YesNo caseFlagValueUpdated;
+    private final String caseSummaryFlagAddedByFullName;
+    private final String caseSummaryFlagAddedByEmail;
```

Comment:
thoughts on putting all of these into their own wrapper class, something like `RedDotEventData` and then in here you can have
```java
@JsonUnwrapped
private final RedDotEventData redDotEventData
```
and then get access to all the values through that. It would mean keeping the mess that is the case data class a bit smaller